maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuilder.java

```diff
@@ -669,6 +669,9 @@ private void initProject( MavenProject project, Map<String, MavenProject> projec
                 else
                 {
                     Artifact parentArtifact = project.getParentArtifact();
+                    if (parentArtifact.getVersion().matches("\\$\\{(revision|sha1|changelist)}")) {
+                        parentArtifact.setVersion(project.getModel().getParent().getVersion());
```

Comment:
First the pattern is not enough if we really allow to use ci friendly in parent cause things like this `${revision}-${changelist}` are also possible which will not work with this pattern..Furthermore an integration test is needed for such things.